draft-ietf-privacypass-architecture.md

```diff
@@ -651,112 +605,101 @@ hit:
+an Issuer, even if the tokens fail later verification checks. This is
 because tokens effectively represent a form of currency that they can
 later redeem for some sort of benefit. The verification checks that are
-put in place are there to ensure that the client does not sacrifice
-their anonymity. However, a client may judge the "monetary" benefit of
+put in place are there to ensure that the Client does not sacrifice
+their anonymity. However, a Client may judge the "monetary" benefit of
 owning tokens to be greater than their own privacy.
 
-Firstly, a client behaving in this way would not be compliant with the
+Firstly, a Client behaving in this way would not be compliant with the
 protocol, as laid out in {{?I-D.ietf-privacypass-protocol}}.
 
 Secondly, acting in this way only affects the privacy of the immediate
-client. There is an exception if a large number of clients colluded to
-accept bad data, then any client that didn't accept would be part of a
+Client. There is an exception if a large number of Clients colluded to
+accept bad data, then any Client that didn't accept would be part of a
 smaller anonymity set. However, such a situation would be identical to
-the situation where the total number of clients in the ecosystem is
+the situation where the total number of Clients in the ecosystem is
 small. Therefore, the reduction in the size of the anonymity set would
 be equivalent; see {{servers}} for more details.
 
 # Security considerations {#security}
 
 We present a number of security considerations that prevent malicious
-clients from abusing the protocol.
+Clients from abusing the protocol.
 
-## Double-spend protection
+## Double-spend protection {#double-spend}
 
-All issuing server should implement a robust storage-query mechanism for
-checking that tokens sent by clients have not been spent before. Such
-tokens only need to be checked for each server individually. But all
-servers must perform global double-spend checks to avoid clients from
+When applicable, all Origins SHOULD implement a robust storage-query
+mechanism for checking that tokens sent by clients have not been spent before.
+Such tokens only need to be checked for each Origin individually. But all
+Origins must perform global double-spend checks to avoid clients from
 exploiting the possibility of spending tokens more than once against
 distributed token checking systems. For the same reason, the global data
 storage must have quick update times. While an update is occurring it
 may be possible for a malicious client to spend a token more than once.
 
 ## Token exhaustion
 
-When a client holds tokens for an server, it is possible for any
-verifier to invoke that client to redeem tokens for that server. This
-can lead to an attack where a malicious verifier can force a client to
-spend all of their tokens for a given server. To prevent this from
+When a Client holds tokens for an Issuer, it is possible for any
```

Comment:
We should mention tokens being scoped to an origin here, or being interactive, as a way to avoid this.